### PR TITLE
don't allow 'limits.memory' to be 0

### LINKED ISSUE #9218: Dont allow limits.memory to be set to zero
Otherwise instance will fail to start.

See https://discuss.linuxcontainers.org/t/failed-to-run-a-container-bad-file-descriptor-failed-to-allocate-new-network-namespace-id/12145/6

### PR DESCRIPTION
**Pull request by masnax**
Closes #9218

### CODE REVIEW

File: shared/instance.go [hunk 1]
```diff
@@ -114,19 +114,27 @@ var InstanceConfigKeysAny = map[string]func(value string) error{
...
 
 		if strings.HasSuffix(value, "%") {
-			_, err := strconv.ParseInt(strings.TrimSuffix(value, "%"), 10, 64)
+			num, err := strconv.ParseInt(strings.TrimSuffix(value, "%"), 10, 64)
 			if err != nil {
 				return err
 			}
 
+			if num == 0 {
+				return fmt.Errorf("Memory limit can't be 0")
```
Comment (stgraber) [inline shared/instance.go:123]: This should probably state `0%` then.

Comment (masnax) [reply shared/instance.go:123]: True indeed.

Comment (stgraber) [reply shared/instance.go]: Needs to be `0%%` actually to avoid issues.

Comment (masnax) [reply shared/instance.go]: Yeah I always forget that :)

Comment (masnax) [reply shared/instance.go]: This actually doesn't seem to work with `fmt.Errorf`, I know @tomponline mentioned we're avoiding using `errors`. Would this be an exception?

Comment (tomponline) [reply shared/instance.go]: Works here https://play.golang.org/p/QE2SsE0jZKk

What version of go are you using?

Comment (masnax) [reply shared/instance.go]: Works if you `fmt.Println` the error afterwards, but I don't think that's how we print errors normally.

Comment (masnax) [reply shared/instance.go]: Oops, accidentally pushed an empty commit :)

Comment (tomponline) [reply shared/instance.go]: Ah you can use the stdlib errors.New() then

Comment (masnax) [reply shared/instance.go]: Should be good to go then, thanks for the input :)

**Pull request merged**